modules/core/src/core/lib/layer.js

```diff
@@ -362,6 +362,71 @@ export default class Layer {
     }
   }
 
+  // This method overwrites the current buffer and is used for multi-picking
+  _clearInstancePickingColor(color) {
+    const {instancePickingColors} = this.getAttributeManager().attributes;
+    const {state: attribute} = instancePickingColors;
+    const {value, size} = attribute;
+
+    const i = this.decodePickingColor(color);
+    value[i * size + 0] = 0;
+    value[i * size + 1] = 0;
+    value[i * size + 2] = 0;
+
+    // TODO: Optimize this to use sub-buffer update!
+    const models = this.getModels();
+    if (models) {
+      models.forEach(model => model.setAttributes({instancePickingColors: attribute}));
+    }
+  }
+
+  // This method overwrites the current buffer and is used for multi-picking
+  _clearPickingColor(color) {
+    const {pickingColors} = this.getAttributeManager().attributes;
+    const attribute = pickingColors.state;
+    const {value} = attribute;
+
+    for (let i = 0; i < value.length; i += 3) {
+      if (value[i + 0] === color[0] && value[i + 1] === color[1] && value[i + 2] === color[2]) {
+        value[i + 0] = 0;
+        value[i + 1] = 0;
+        value[i + 2] = 0;
+      }
+    }
+
+    // TODO: Optimize this to use sub-buffer update!
+    const models = this.getModels();
+    if (models) {
+      models.forEach(model => model.setAttributes({pickingColors: attribute}));
+    }
+  }
+
+  // This method figures out if we use instance colors or not
+  // and calls _clearInstancePickingColor or _clearPickingColor
+  clearPickingColor(color) {
```

Comment:
Maybe for a second PR: Would it be a better interface to have this method take an object index (or even list of indices) and keep all the pickingColor translation knowledge inside here?

Reply:
Maybe...